lib/assocPath/index.js

```diff
@@ -0,0 +1,17 @@
+import _curry3 from '../_internal/_curry3'
+
+export default _curry3(function assocPath(path, value, obj) {
+  var pathObj = path.reverse().reduce(function(acc, item, i) {
+    i === 0
+      ? acc[item] = value
+      : acc = { [item]: acc }
+    return acc
+  }, {})
+
+  return Object.keys(obj).reduce(function(acc, key, i, arr) {
+    pathObj.hasOwnProperty(key)
+      ? acc[key] = pathObj[key]
+      : acc[key] = obj[key]
```

Comment:
should it be a deep copy of original object or shallow?
because side effects will be if you want to change the value of our object
Example:
```
var a = { b: { c: { d: 1 } } };
var copyA = assocPath(['b', 'c', 'd'], 2, a)
var copyCopyA = assocPath(['b', 'c'], { d: 3 }, copyA)
expect(a.b.c.d).toEqual(copyCopyA.b.c.d);
```